### ISSUE TITLE
Unused imports

### ISSUE THREAD

**Issue by Timmmm** [eslint/eslint#13650]
**Please describe what the rule should do:**

Warn if you import something using ES6 import syntax and don't use it.

**What new ECMAScript feature does this rule relate to?**

ES6 modules.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[*] Other: Code cleanup

**Provide 2-3 code examples that this rule will warn about:**

```js
import foo from "bar";
import { baz } from "qux";

// foo or baz are not used.
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It's a basic rule. There is [a plugin that does this](https://www.npmjs.com/package/eslint-plugin-unused-imports) but it's such a basic thing that I think it should really be part of the core experience, in the same way that reporting unused variables is. In fact, reporting unused variables does handle this if you use the old `require()` module system.

**Are you willing to submit a pull request to implement this rule?**

No, but the code is already written so I assume it wouldn't be too difficult.

**Comment by mdjermanovic** [comment]
Hi @Timmmm, thanks for the issue!

The core `no-unused-vars` rule already reports unused imports, please check [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBlcnJvciAqL1xuXG5pbXBvcnQgZm9vIGZyb20gXCJiYXJcIjtcbmltcG9ydCB7IGJheiB9IGZyb20gXCJxdXhcIjtcblxuLy8gZm9vIG9yIGJheiBhcmUgbm90IHVzZWQuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19).

**Comment by Timmmm** [comment]
Ah yeah, my mistake. I had turned it off a while ago due to a bug when importing Typescript interfaces ironically! I guess that is an issue with eslint-typescript though.
